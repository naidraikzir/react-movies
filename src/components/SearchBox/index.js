import React from 'react';
import {
  InputGroup,
  Input,
  InputRightElement,
  Button,
} from '@chakra-ui/react';
import { CloseIcon } from '@chakra-ui/icons';

const Search = ({
  value,
  onInput,
  onEnter,
  onReset,
}) => (
  <InputGroup>
    <Input
      value={value}
      onInput={onInput}
      onKeyUp={evt => evt.key === 'Enter' && onEnter(evt)}
      shadow="lg"
      placeholder="Type and press enter to search movies..."
    />
    {value.length && <InputRightElement p="5">
      <Button
        borderRadius="full"
        size="xs"
        onClick={onReset}
      >
        <CloseIcon color="gray.500" />
      </Button>
    </InputRightElement>}
  </InputGroup>
);

export default Search;
