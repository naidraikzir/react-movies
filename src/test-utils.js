import React from 'react';
import { render } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { ChakraProvider, theme } from '@chakra-ui/react';
import reducer from 'store/reducer';

const customRender = (
  ui,
  {
    initialState,
    store = createStore(reducer, initialState),
    ...options
  } = {}
) => {
  const AllProviders = ({ children }) => (
    <Provider store={store}>
      <ChakraProvider theme={theme}>{children}</ChakraProvider>
    </Provider>
  );
  return render(ui, { wrapper: AllProviders, ...options });
}

export { customRender as render };
