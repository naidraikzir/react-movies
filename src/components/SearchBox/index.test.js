import React from 'react';
import { fireEvent } from '@testing-library/react';
import { render } from 'test-utils';
import SearchBox from '.';

test('match snapshot', () => {
  const { container } = render(<SearchBox />);
  expect(container.firstChild).toMatchSnapshot();
});

test('calls onEnter prop when user press enter in the input', () => {
  const mockOnEnter = jest.fn();
  const { container } = render(<SearchBox onEnter={mockOnEnter} />);
  fireEvent.keyUp(container.querySelector('[name="search"]'), { key: 'Enter' });
  expect(mockOnEnter).toHaveBeenCalled();
});

test('renders reset button if has value', () => {
  const { container } = render(<SearchBox value="search..." />);
  expect(container.getElementsByTagName('button')).toHaveLength(1);
});

test('not render reset button if has no value', () => {
  const { container } = render(<SearchBox />);
  expect(container.getElementsByTagName('button')).toHaveLength(0);
});

test('calls onReset on reset button clicked', () => {
  const mockOnReset = jest.fn();
  const { container } = render(<SearchBox
    value="search"
    onReset={mockOnReset}
  />);
  fireEvent.click(container.getElementsByTagName('button')[0]);
  expect(mockOnReset).toHaveBeenCalled();
});
