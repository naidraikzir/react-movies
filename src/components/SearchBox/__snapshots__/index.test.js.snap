// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  class="chakra-input__group css-4302v8"
>
  <input
    class="chakra-input css-1yyyhya"
    name="search"
    placeholder="Type and press enter to search movies..."
    value=""
  />
</div>
`;
